port_ocean/core/integrations/mixins/sync_raw.py

```diff
@@ -135,42 +138,41 @@ async def _register_resource_raw(
         resource: ResourceConfig,
         results: list[dict[Any, Any]],
         user_agent_type: UserAgentType,
+        parse_all: bool = False,
     ) -> list[Entity]:
-        objects_diff = await self._calculate_raw(
-            [
-                (
-                    resource,
-                    {
-                        "before": [],
-                        "after": results,
-                    },
-                )
-            ]
-        )
+        objects_diff = await self._calculate_raw([(resource, results)], parse_all)
 
-        entities_after: list[Entity] = objects_diff[0]["after"]
+        entities_after: list[Entity] = objects_diff[0]["passed"]
         await self.entities_state_applier.upsert(entities_after, user_agent_type)
+
+        # If an entity didn't pass the JQ selector, we want to delete it if it exists in Port
+        for entity_to_delete in objects_diff[0]["failed"]:
+            is_owner = (
+                await ocean.port_client.does_integration_has_ownership_over_entity(
+                    entity_to_delete, user_agent_type
+                )
+            )
+            if not is_owner:
+                logger.info(
+                    f"Entity {entity_to_delete.identifier} hasn't been created by this integration, "
+                    f"skipping deletion..."
+                )
+                continue
+            await self.entities_state_applier.delete(
+                objects_diff[0]["failed"], user_agent_type
```

Comment:
this log might be confusing, as it is possible that the entity doesn't exists in port which in that case will return `is_owner=False` as well.